src/main/java/formflow/library/ScreenController.java

```diff
@@ -131,6 +131,7 @@ ModelAndView postScreen(
     // if there's already a session
     if (submission.getId() != null) {
       submission.mergeFormDataWithSubmissionData(formSubmission);
+      handleBeforeSaveAction(getScreenConfig(flow, screen), submission);
```

Comment:
I _think_ you want this outside of whether or not a submission already exists.  This way, the action occurs regardless of whether or not previous data existed.  

I was thinking that maybe this could be done before this if/else statement, though I can see this is tricky with the merging of form data needing to happen first (if submission data already exists).  Given that, I do think the `handleBeforeSaveAction` may need to be called in the else as well.  Esp. if the before save action only depends on currently entered data - we'd still want the action to happen, even if no previous data existed.

Reply:
This is subtle, too, as a test would miss this unless the action were on the first screen of inputs.